### PR TITLE
(QENG-1869) create beaker host.rb acceptance tests

### PR DESCRIPTION
**Pull request by anodelman**
- tests that exercise basic host.rb functionality

- discovered during testing
  * consistency of how dirs are recursively copied in do_scp_to
  * how ignored files/dirs are treated

### CODE REVIEW

**Comment by anodelman** [comment]
retest this please

**Comment by anodelman** [comment]
Failures unrelated to patch - passes beaker-rspec module testing integration.

**Comment by kevpl** [comment]
@anodelman waiting for you to be available IRL before merging this...

File: acceptance/fixtures/module/spec/spec_helper_acceptance.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  if default.is_pe?; then
+    install_pe;
+  else
+    install_puppet(foss_opts);
+  end
+
+  hosts.each do |host|
+    unless host.is_pe?
+      on host, "/bin/echo '' > #{host['hieraconf']}"
```
Comment (joshcooper) [inline acceptance/fixtures/module/spec/spec_helper_acceptance.rb]: `host.puppet('hiera_config')`?

File: lib/beaker/host.rb [hunk 2]
```diff
@@ -212,8 +212,13 @@ def ip
     #@return [Boolean] true if x86_64, false otherwise
     def determine_if_x86_64
       if is_cygwin?
-        result = exec(Beaker::Command.new("arch | grep x86_64"), :acceptable_exit_codes => (0...127))
-        result.exit_code == 0
+        if self[:platform] =~ /osx|solaris/
```
Comment (joshcooper) [inline lib/beaker/host.rb]: shouldn't this be outside the `if is_cygwin?`?

Comment (anodelman) [inline lib/beaker/host.rb]: is_cygwin identified the box as being not-powershell windows - anything that is not in the bucked 'powershell' is identified as not being cygwin.

**Comment by anodelman** [comment]
retest this please

**Comment by kevpl** [comment]
failures caused by timeout, filed the issue to fix that separately.  will retest this

**Comment by kevpl** [comment]
@puppetlabs-jenkins retest this please

**Pull request merged**